Content.Shared/GameObjects/EntitySystems/Atmos/SharedGasTileOverlaySystem.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Map;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.GameObjects.EntitySystems.Atmos
+{
+    public abstract class SharedGasTileOverlaySystem : EntitySystem
+    {
+        public const byte ChunkSize = 8;
+        
+        /// <summary>
+        ///     How frequently we can send out new data
+        /// </summary>
+        protected const float UpdateTime = 0.16f;
+        protected float AccumulatedFrameTime;
+
+        // Probably need a cvar for this.
+        public const float UpdateRange = 18;
+        
+        public static MapIndices GetGasChunkIndices(MapIndices indices)
+        {
+            return new MapIndices((int) Math.Floor((float) indices.X / ChunkSize) * ChunkSize, (int) MathF.Floor((float) indices.Y / ChunkSize) * ChunkSize);
+        }
+        
+        [Serializable, NetSerializable]
+        public struct GasData
+        {
+            public int Index { get; set; }
+            public float Opacity { get; set; }
```

Comment:
Is it possible to change this to like `byte` index and `byte` opacity? It would cut down the amount of bandwidth necessary for these updates tremendously.